src/WatchDogs.Infrastructure.FakeSource/Watcher.cs

```diff
@@ -16,13 +17,15 @@ public class Watcher : IWatcher
     private readonly CancellationTokenSource _cts = new();
     private readonly IFakeTradeGenerator _dataGenerator;
     private readonly ILogger _logger;
+    private readonly DataInserter _dataInserter;
 
 
 
-    public Watcher(TimeSpan interval, IFakeTradeGenerator dataGenerator)
+    public Watcher(TimeSpan interval, IFakeTradeGenerator dataGenerator, DataInserter dataInserter)
```

Comment:
See? It is tightly coupled to DataInserter

Reply:
Done.